x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/advanced_detector_modal.tsx

```diff
@@ -251,7 +251,7 @@ export const AdvancedDetectorModal: FC<Props> = ({
               <EuiComboBox
                 singleSelection={{ asPlainText: true }}
                 options={currentFieldOptions}
-                selectedOptions={createSelectedOptions(fieldOption, currentFieldOptions)}
+                selectedOptions={createSelectedOptions(fieldOption)}
```

Comment:
For the sample logs data set, why is `hour_of_day` appearing as a suitable option for a `lat_long` job? Shouldn't it just show geo fields?

![image](https://user-images.githubusercontent.com/7405507/174316255-a5664c4b-8d33-4417-8e48-16e030dfc83a.png)

Reply:
runtime fields are appended to the dropdowns regardless of type.
To change this will need to be a future enhancement. I've created an [issue](https://github.com/elastic/kibana/issues/134678) for it.

Reply:
Looks like `keyword` runtime fields don't have this issue. Numeric types (tried `Long` and `Double`) do exhibit this issue.